File: frontend/src/types/application.ts
export type Status =
  | "applied"
  | "phone_screen"
  | "technical_interview"
  | "final_interview"
  | "offer"
  | "rejected"
  | "withdrawn";

export interface JobApplication {
  id: number;
  company: string;
  position: string;
  location: string;
  jobUrl: string;
  status: Status;
  appliedDate: string;
  notes: string;
  contactPerson: string;
  salaryMin: number | null;
  salaryMax: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface ListFilter {
  search: string;
  statuses: Status[];
  sortBy: "appliedDate" | "company" | "status";
  sortDir: "asc" | "desc";
}

export interface ApplicationInput {
  company: string;
  position: string;
  location: string;
  jobUrl: string;
  status: Status;
  appliedDate: string;
  notes: string;
  contactPerson: string;
  salaryMin: number | null;
  salaryMax: number | null;
}

export interface DashboardStats {
  total: number;
  active: number;
  offers: number;
  rejected: number;
}

export interface StatusCount {
  status: Status;
  count: number;
}

export interface TimeSeriesPoint {
  date: string;
  count: number;
}

File: internal/models/application.go
package models

import "time"

type Status string

const (
	StatusApplied            Status = "applied"
	StatusPhoneScreen        Status = "phone_screen"
	StatusTechnicalInterview Status = "technical_interview"
	StatusFinalInterview     Status = "final_interview"
	StatusOffer              Status = "offer"
	StatusRejected           Status = "rejected"
	StatusWithdrawn          Status = "withdrawn"
)

var StatusOrder = []Status{
	StatusApplied,
	StatusPhoneScreen,
	StatusTechnicalInterview,
	StatusFinalInterview,
	StatusOffer,
	StatusRejected,
	StatusWithdrawn,
}

type JobApplication struct {
	ID            uint      `gorm:"primaryKey"       json:"id"`
	Company       string    `gorm:"not null;index"   json:"company"`
	Position      string    `gorm:"not null"         json:"position"`
	Location      string    `json:"location"`
	JobURL        string    `json:"jobUrl"`
	Status        Status    `gorm:"not null;index"   json:"status"`
	AppliedDate   time.Time `gorm:"not null;index"   json:"appliedDate"`
	Notes         string    `json:"notes"`
	ContactPerson string    `json:"contactPerson"`
	SalaryMin     *int      `json:"salaryMin"`
	SalaryMax     *int      `json:"salaryMax"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}
